### PR TITLE
Branch bug fix

### PR DESCRIPTION
**Pull request by janald99**
Fix match index issue #213 where index out of range did not throw an exception previously

#251 where warning was placed at sort instead of clear

#242 user guide updated to say u cannot match null property, and added the code accordingly for that

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -153,10 +153,26 @@ Format: `add-ptb INDEX l/LOCATION pr/PRICE_RANGE h/HOUSE_TYPE`
 
 * Adds a new property that the buyer at `INDEX` is hoping to buy. The index refers to the index number shown in the displayed buyer list. The index **must be a positive whole number** 1, 2, 3, …​
 * The order of the inputs can be in any order.
+
+* None of the inputs can be empty. E.g. Typing `l/` instead of `l/Bishan` will result in an error.
+* The `HOUSE_TYPE` can be defined as any of the following:
```
Comment (Jacky142857) [inline docs/UserGuide.md:158]: any of the followings (plural?)

File: docs/UserGuide.md [hunk 2]
```diff
@@ -178,7 +194,8 @@ Examples:
 
 Function: Display potential sellers by matching the demand of a buyer. Now, you can easily find properties that buyers are interested in!
 
-* Want to match using just a certain criterion? Fret not! Here are the different match commands:
+* Note: You can match a buyer only if a property has been added!
```
Comment (Jacky142857) [inline docs/UserGuide.md:197]: Maybe it's better to say 'You can match a buyer only if his/her property has been added!' or 'a property of him/her'

Comment (janald99) [reply docs/UserGuide.md:197]: Will update accordingly!

File: src/main/java/seedu/address/logic/commands/MatchCommand.java [hunk 3]
```diff
@@ -33,8 +34,17 @@ public MatchCommand(Index index) {
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         ObservableList<Buyer> buyerList = model.getFilteredBuyerList();
+
+        if (index.getZeroBased() >= buyerList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_CLIENT_DISPLAYED_INDEX);
```
[the remark this answers is not in the document]
Comment (janald99) [reply src/main/java/seedu/address/logic/commands/MatchCommand.java:39]: Oh yes, will change this!

File: docs/UserGuide.md [hunk 4]
```diff
@@ -159,10 +159,26 @@ Format: `add-ptb INDEX l/LOCATION pr/PRICE_RANGE h/HOUSE_TYPE`
...
+  * `condominium` or `condo`
+  * `colonia`
+  * `hdb` or `hdb_flat`
+  * `semi-detached` or `semi-d` or `semidetached` or `semid`
+  * `mansion`
+* Any other `HOUSE_TYPE` will not be accepted.
+* The default `HOUSE_TYPE` after adding a client is none of the above. After a property has been added, you cannot reset it back to this default.
+* `LOCATION` can be any non-empty input, but do use appropriate locations for your own utility. E.g. `Bishan` or `Marymount`
+* The `PRICE_RANGE` is specified in the following format: `lower,upper`.
 * The `PRICE_RANGE` must be a valid **non-negative integer** with `lower` being less than or equal to `upper`.
```
Comment (cwnm) [inline docs/UserGuide.md:177]: Should it be price range inputs must be valid integer?

File: docs/UserGuide.md [hunk 5]
```diff
@@ -159,10 +159,26 @@ Format: `add-ptb INDEX l/LOCATION pr/PRICE_RANGE h/HOUSE_TYPE`
...
 * The `PRICE_RANGE` can include `0` since the buyer might want to try their luck and see if anyone is selling their property for free.
 * Do use appropriate `LOCATION` for your own utility. E.g. `Bishan` or `Marymount`. The application will not check if it is an actual location in Singapore or elsewhere.
```
Comment (cwnm) [inline docs/UserGuide.md:179]: Is the first part a repeat of line 175?

**Pull request merged**